### DISCUSSION TITLE
Odometer reading - scan live or from camera roll

### DISCUSSION THREAD

**Discussion by Jakayus** [mikaelacaron/Basic-Car-Maintenance#186] [Ideas]
I am prone to typos so I was thinking of a way to input the odometer manually without having to type anything. I am wondering if it would be worthwhile to try and scan the odometer using the camera, either live or through picking out a picture in the camera roll.

The live scan would be hard to verify with the simulator 😅 But at least a picture from camera roll could help.

Inspired by the camera roll idea from #122 and also the odometer manual/auto discussion from #145, so giving credit where credit is due.

Also I haven't reviewed the latest OdometerView status so my apologies if this is already been discussed or worked.

**Comment by mikaelacaron** [comment]
Yeah as you said it's noted in scenario 2 of [this post](https://github.com/mikaelacaron/Basic-Car-Maintenance/discussions/145#discussion-5723833)!

I think this totally could happen, and would be an awesome feature, and could use Firebase ML, but I'm not yet 100% sure if people can run on device, because the app uses Sign in with Apple, which makes it so that you need that capability set up properly.

If you can figure out how to run on device (after properly setting up the project), that would be the first step.

Technically I think you'd have to make a bundle ID, and add the right capability in App Store Connect, to do this, but I'm not completely sure

**Comment by Jakayus** [comment]
Oops yeah, looks like I skimmed past the camera points in that discussion.

I'll experiment a bit this week and see if I can find a way to run on device! Scenario 2 would be a cool feature to add. 

Thank you for the starting point suggestions, those are good thoughts to keep in mind. If I make any progress I'll respond here.

**Reply by mikaelacaron** [reply]
Can you actually respond over on that other thread? Writing a new response below, and then we can thread discussion there, because it's all about the same thing

Thanks!
